### PR TITLE
chore: Upgrades Next.js to 13.4.6

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Upgrades Next.js to 13.4.6

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- [ ] yarn build
- [ ] yarn e2e

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by emrysal** [comment]
I fixed the tests;

Turns out NextJS 13.3.0 has an as far as I can tell undocumented breaking change; at least on localhost and staging. API endpoints no longer receive a request body as part of DELETE calls. I've looked up where we do this (calendar switch, cancelation) and fixed these endpoints.

https://github.com/vercel/next.js/discussions/48072
https://github.com/vercel/next.js/issues/48096
https://github.com/vercel/next.js/issues/49353

**Review by emrysal — APPROVED** [review]

File: packages/features/calendars/CalendarSwitch.tsx [hunk 1]
```diff
@@ -47,12 +47,11 @@ const CalendarSwitch = (props: ICalendarSwitchProps) => {
...
         }
       } else {
-        const res = await fetch("/api/availability/calendar", {
+        const res = await fetch("/api/availability/calendar?" + new URLSearchParams(body), {
           method: "DELETE",
           headers: {
             "Content-Type": "application/json",
           },
-          body: JSON.stringify(body),
         });
```
Comment (roae) [inline packages/features/calendars/CalendarSwitch.tsx:55]: @emrysal I'm wondering why is this needed? is this related to the NextJS update?

Comment (emrysal) [reply packages/features/calendars/CalendarSwitch.tsx:55]: See my comment on this PR that elaborates why; tl;dr; yes.

Comment (roae) [reply packages/features/calendars/CalendarSwitch.tsx:55]: Ooh! I got it, thanks!

**Review by roae — APPROVED** [review]
It works on my end!

**Review by ericrommel — APPROVED** [review]
Working fine from my local
